Services/CO.CDP.Organisation.WebApi/UseCase/ReviewOrganisationUseCase.cs

```diff
@@ -0,0 +1,25 @@
+using CO.CDP.Organisation.WebApi.Model;
+using CO.CDP.OrganisationInformation.Persistence;
+
+namespace CO.CDP.Organisation.WebApi.UseCase;
+public class ReviewOrganisationUseCase(IOrganisationRepository organisationRepository, IPersonRepository personRepository)
+    : IUseCase<ReviewOrganisation, Boolean>
+{
+    public async Task<Boolean> Execute(ReviewOrganisation command)
+    {
+        var organisation = await organisationRepository.Find(command.OrganisationId) ?? throw new UnknownOrganisationException($"Unknown organisation {command.OrganisationId}.");
+        var person = await personRepository.Find(command.approvedById) ?? throw new UnknownPersonException($"Unknown person {command.approvedById}.");
+
+        // Organisation is left in a pending state until user approves it.
+        if (command.Approved)
+        {
+            organisation.ApprovedOn = DateTimeOffset.UtcNow;
```

Comment:
@jakzal At the moment this `ApprovedOn` is what marks the buyer as approved. After the buyer is approved what is it that they will be able to do? Is there somewhere where I need to apply a check that the buyer is approved before actions or something else can be done, or is this functionality to be added at a later date?

Reply:
This is a question for BAs. We need a ticket for this in the backlog imo. What should the buyer be able to do once they're approved? What shouldn't they be able to do before they're approved? Do we return them with a buyer role from our APIs before they're approved?